JSONPointer.java

```diff
@@ -74,6 +101,10 @@ public JSONPointer(String pointer) {
         }
     }
 
+    public JSONPointer(List<String> refTokens) {
```

Comment:
I we want this could either be a protected constructor, or the List should be copied so the Pointer can't be modified outside of the class.

this.refTokens = new ArrayList<>(refTokens);